admin/resources/js/components/form/Submit.tsx

```diff
@@ -0,0 +1,27 @@
+import React from "react";
+import { useFormState } from "react-hook-form";
+
+export const Submit: React.FunctionComponent<{
+  text?: string;
+  notDirtyText?: string;
+  isSubmittingText?: string;
+}> = ({ text, notDirtyText, isSubmittingText }) => {
+  const _text = text ?? "Submit";
+  const _notDirtyText = notDirtyText ?? "Submitted";
+  const _isSubmittingText = isSubmittingText ?? "Submitting";
+  const { isDirty, isSubmitting } = useFormState();
+
+  const currentText = isSubmitting
+    ? _isSubmittingText
+    : !isDirty
+    ? _notDirtyText
+    : _text;
```

Comment:
This is pretty hard to read 😄 maybe we should use an if statement or switch case instead? That's probably why that no nested ternary operator rule exists. Also, the button initially reads "Submitted" (on first open), when it should read "Submit".

Reply:
There's also some inconsistencies with the button with the two stories: normal submit, and the failing submit.

Reply:
Alright, I guess I just needed the sober second thought from you, I'll restore the eslint rules 😆 .

Initially reading "Submitted" is intended. You haven't changed anything, so there's no reason to submit.

For the inconsistency, do you mean how after a failure the button still reads "Submit"? That's also intended. Submission didn't work correctly, so the user should be able to try again. This will be more clear after we have errors displaying, I think.

Reply:
Yea I was wrong about the inconsistencies with the two forms. They are working as expected. However, I added a change to check if it has been submitted so now it will initially read "Submit".